packages/sdk-core/src/proposals/ProposalKinds.ts

```diff
@@ -1,8 +1,192 @@
+import type { TransactionV2 } from '@polkadot/types/interfaces/eth';
+
+import { BE } from '@webb-tools/sdk-core/proposals/index';
+
+import { TypeRegistry } from '@polkadot/types';
 import { hexToU8a, u8aToHex } from '@polkadot/util';
 
 import { ProposalHeader } from './ProposalHeader.js';
 import { ResourceId } from './ResourceId.js';
 
+export interface IEVMProposal {
+  readonly chainId: number;
+  readonly nonce: number;
+  readonly tx: TransactionV2
+}
+
+export class EVMProposal implements IEVMProposal {
+  readonly chainId: number;
+  readonly nonce: number;
+  readonly tx: TransactionV2;
+
+  constructor (
+    chainId: number,
+    nonce: number,
+    tx: TransactionV2
+  ) {
+    this.chainId = chainId;
+    this.nonce = nonce;
+    this.tx = tx;
+  }
+
+  fromBytes (bytes: Uint8Array): EVMProposal {
+    const reg = new TypeRegistry();
+    const tx = reg.createType<TransactionV2>('TransactionV2', bytes);
+    let nonce: number;
+    let chainId = 0;
+
+    switch (tx.type) {
+      case 'Legacy':{
+        const txValue = tx.asLegacy;
+
+        nonce = txValue.nonce.toNumber();
+        chainId = 0;
+      }
+
+        break;
```

Comment:
Aren't these breaks in the wrong place? Shouldn't they be inside each `case`?

Reply:
Not sure why but this happens when I run `yarn lint` will check it more

Reply:
switched to if instead